src/Components/HOCs/withAuthentication.tsx

```diff
@@ -1,136 +1,91 @@
-import React, { useState, useEffect, useRef, useCallback } from "react";
+import React, { useState, useCallback } from "react";
 import { Button, Modal } from "@material-ui/core";
 import { Authentication, PermissionType, User } from "@mov-ai/mov-fe-lib-core";
+import { Emit, makeSub } from "../../Utils/Sub";
+import useSub from "../../hooks/useSub";
 import LoginForm from "../LoginForm/LoginForm";
 import LoginPanel from "../LoginForm/LoginPanel";
-import jwtDecode from "jwt-decode";
 import i18n from "../../i18n/i18n.js";
 
-export default function withAuthentication<P extends object>(
-  WrappedComponent: React.ComponentType<P>,
-  appName: PermissionType | string
+interface LoginData {
+  username: string;
+  password: string;
+  remember: any;
+  selectedProvider: any; 
+}
+
+interface LoginSub {
+  loggedIn: boolean,
+  currentUser: any,
+  loading: boolean,
+  providers: { domains: string[] },
+}
+
+export
+const loggedOutInfo = {
+  loggedIn: false,
+  currentUser: null,
+  loading: false,
+  providers: { domains: [] },
+};
```

Comment:
we decided (long ago before you joined) that constants like this should be in all caps with underscores, for better readability. Also, if they are going to be used in more than one place (that's why you're exporting it), then it should be on the constants file.

This is not mandatory for this PR. Just if you have the time and agree. Since it was decided before you, it's logic that you are unaware. If it's worth it I might try and make a presentation on these kind of things that were decided before.

```suggestion
const LOGGED_OUT_INFO = {
  LOGGED_IN: false,
  CURRENT_USER: null,
  LOADING: false,
  PROVIDERS: { domains: [] },
};
```

Reply:
Ok. But this not a usual constants definition. Its sort of the initial state for react state.
But this is for subscriptions.

Of course, we can commit to all caps for state.
But then you are going to get situations like these:
```js
function Component() {
  const { LOGGED_IN } = useSub(authSub);
  ...
}
```
That would be ugly in my opinion. What do you think?